### DISCUSSION TITLE
How can I change the color of a row with a column condition?

### DISCUSSION THREAD

**Discussion by RonalSilve** [KevinVandy/material-react-table#340] [Q&A] [answered]
I want to avoid creating a status column by giving a color for each value the column has.

**Comment by KevinVandy** [comment, marked as answer]
See both the [Column Instance API Doc](https://www.material-react-table.com/docs/api/column-instance-apis)s and the [Row Instance API docs](https://www.material-react-table.com/docs/api/row-instance-apis). You can pass props to each row conditionally with the `muiTableBodyRowProps` prop like so

```jsx
<MaterialReactTable
  muiTableBodyRowProps={({ column, row }) => ({
    //props, conditional on column and row apis
  })
/>
```

**Reply by RonalSilve** [reply]
Thanks for the help, this way I was able to solve the problem with the suggestion you gave me:

```
<MaterialReactTable
      columns={columns}
      data={data}
      muiTableBodyRowProps={({ row }) => ({
        style:{
          backgroundColor: row.original.city === 'Lincoln' ? 'blue' : 'white'
        }
      })}
    />
```

**Reply by mcadotsch** [reply]
@KevinVandy 
We use muiTableBodyRowProps to change conditionally the color of a row.
Now with the columnPinning enabled for the action column, it seems that the colorisation of the action cell stoped to work:

```
columnPinning: { right: ['mrt-row-actions'] }

muiTableBodyRowProps = ({ row }) => ({
    hover: false,
    sx: () => ({
      backgroundColor: row.original.stepColor
    }),
  });
```

I tried to fix it with this change but it did not help:

```  
displayColumnDefOptions: {
    'mrt-row-actions': {
      size: 10,
      muiTableBodyCellProps: { align: 'left' },
      muiTableHeadCellProps: { align: 'left' },
      Cell: ({row, cell})=>(
        <div style={{backgroundColor: "red"}}>{cell.getValue()}</div>
      ),
    }
  },
```

Any idea?

**Reply by KevinVandy** [reply]
Target the td instead of the tr for the background color

**Reply by mcadotsch** [reply]
How do you mean?
As soon as i add the column pinning option the background color changes to default for the action column.

![image](https://github.com/KevinVandy/material-react-table/assets/44357848/981a174c-683c-4193-8c10-ef09fdfd2bd6)
